src/panels/AssetsFolderPanel/AssetsEntry.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Button,
+  CheckIcon,
+  Group,
+  Loader,
+  Text,
+  ThemeIcon,
+  Tooltip
+} from '@mantine/core';
+
+import { useOpenSpaceApi } from '@/api/hooks';
+import { CopyToClipboardButton } from '@/components/CopyToClipboardButton/CopyToClipboardButton';
+import { FileTextIcon } from '@/icons/icons';
+import { IconSize } from '@/types/enums';
+import { eventBus } from '@/util/eventBus';
+
+import { Asset } from './types';
+
+interface Props {
+  asset: Asset;
+}
+
+export function AssetsEntry({ asset }: Props) {
+  const luaApi = useOpenSpaceApi();
+  const [loaded, setLoaded] = useState<boolean | undefined>(undefined);
+  const [loading, setLoading] = useState<boolean>(false);
+  const { t } = useTranslation('panel-assets', { keyPrefix: 'asset-entry' });
+
+  useEffect(() => {
+    async function handleAssetEvent() {
+      const isNowLoaded = await luaApi?.asset.isLoaded(asset.path);
+      // Only need to update state if we finished loading this asset
+      if (isNowLoaded) {
+        setLoading(false);
+        setLoaded(true);
+        eventBus.off('AssetLoadingFinished', handleAssetEvent);
```

Comment:
Seems to me these strings could benefit from being put into an object or something to not have "loose hardcoded strings"